src/Components/NavBar/NavItem.tsx

```diff
@@ -16,6 +16,7 @@ interface NavItemProps extends BoxProps {
   className?: string
   href?: string
   onClick?: () => void
+  onHover?: boolean
```

Comment:
Props with the `onSomething` form are always callback functions since the `onSomeEvent` form implies some action is happening, like a hover or a mouse click (etc).